src/ddmd/aliasthis.d

```diff
@@ -41,14 +41,25 @@ extern (C++) final class AliasThis : Dsymbol
     override Dsymbol syntaxCopy(Dsymbol s)
     {
         assert(!s);
-        /* Since there is no semantic information stored here,
-         * we don't need to copy it.
-         */
-        return this;
+        return new AliasThis(loc, ident);
```

Comment:
Yes. I actually don't think with the current state of the compiler it is possible to trigger the bug, but it is a latent problem that may show up later. I discovered it because I am trying to remove dependence on the state of `_scope` being used to determine if `semantic()` has been done or not. By sharing the `AliasInstance` objects, this PR was failing the test suite when I switched it to relying on `sematicRun`.